metabase/metabase
# Allow CSV Uploads for non-actions-enabled databases / handle permissions

Resolves https://github.com/metabase/metabase/issues/31261

## Backend
* Add `include_only_uploadable?` flag to database endpoint (useful for populating the list of DBs in /admin/settings/uploads).
* Only show databases to users with "Manage Database" permission (EE-only)
* Only let the uploads database be set by those users.


## Frontend
- stop filtering out non-actions-enabled databases on the frontend
- use new `include_only_uploadable` flag to load a list of upload-compatible databases
- upload copy and error messaging

New no-valid-db state

![Screen Shot 2023-06-05 at 10 01 55 AM](https://github.com/metabase/metabase/assets/30528226/38881f56-7b5e-42de-b6df-97842bfb47df)

How to test:
- go to admin > settings > uploads
- see that by default the sample DB is available as an upload destination, even if actions are disabled
- see that if you delete the sample DB and any other compatible DBs (postgres, h2, mysql), you get the above error message informing about CSV-upload-compatible DBs
